src/cmd/cli/command/commands.go

```diff
@@ -859,31 +831,51 @@ var composeUpCmd = &cobra.Command{
 			return errors.New("no services being deployed")
 		}
 
-		serviceInfos := deploy.GetServices()
-		printPlaygroundPortalServiceURLs(serviceInfos)
+		printPlaygroundPortalServiceURLs(deploy.Services)
 
 		if detach {
-			term.Info("Done.")
+			term.Info("Detached.")
 			return nil
 		}
 
-		tailCtx, cancelTail := context.WithCancel(ctx)
+		ctx, cancelTail := context.WithCancelCause(cmd.Context())
+		defer cancelTail(nil) // to cancel waitServiceState
+
+		errSuccess := errors.New("deployment succeeded")
 
-		var wg sync.WaitGroup
-		wg.Add(1)
 		go func() {
-			defer wg.Done()
-			// show users the current streaming logs
-			if err := startTailing(tailCtx, deploy.Etag, since); err != nil {
-				var cerr *cli.CancelError
-				if !errors.As(err, &cerr) {
-					term.Debugf("failed to start tailing: %v", err)
+			services := make([]string, len(deploy.Services))
+			for i, serviceInfo := range deploy.Services {
+				services[i] = serviceInfo.Service.Name
+			}
+
+			if err := waitServiceState(ctx, defangv1.ServiceState_SERVICE_COMPLETED, services); err != nil {
+				if errors.Is(err, errDeploymentFailed) {
+					cancelTail(err)
+				} else if !errors.Is(err, context.Canceled) {
+					term.Warnf("failed to wait for service status: %v", err)
 				}
+			} else {
+				cancelTail(errSuccess)
 			}
 		}()
 
-		if err := waitServiceState(ctx, defangv1.ServiceState_SERVICE_COMPLETED, serviceInfos); err != nil && !errors.Is(err, context.Canceled) {
-			if errors.Is(err, ErrDeploymentFailed) {
+		// show users the current streaming logs
+		services := "all services"
+		if deploy.Etag != "" {
+			services = "deployment ID " + deploy.Etag
+		}
+
+		term.Info("Tailing logs for", services, "; press Ctrl+C to detach:")
+		tailParams := cli.TailOptions{
+			Etag:  deploy.Etag,
+			Since: since,
+			Raw:   false,
+		}
+
+		// blocking call to tail
+		if err := cli.Tail(ctx, client, tailParams); err != nil {
+			if errors.Is(context.Cause(ctx), errDeploymentFailed) {
 				term.Warn("Deployment FAILED. Service(s) not running.")
 
 				if !nonInteractive {
```

Comment:
non-interactive prompt in main thread now, which fixes the issue @raphaeltm was having